core/server/lib/members/static/gateway/bundle.js

```diff
@@ -23,9 +23,76 @@
         };
     }
 
+    function isTokenExpired(token) {
+        try {
+            const [header, claims, signature] = token.split('.');
+
+            const parsedClaims = JSON.parse(atob(claims.replace('+', '-').replace('/', '_')));
+
+            const expiry = parsedClaims.exp * 1000;
+            const now = Date.now();
+
+            const nearFuture = now + (30 * 1000);
```

Comment:
This means that we consider a token expired if it will expire in 30 seconds, this should be more than enough to account for network lag.